pymc/logprob/censoring.py

```diff
@@ -237,3 +240,149 @@ def round_logprob(op, values, base_rv, **kwargs):
     from pymc.math import logdiffexp
 
     return logdiffexp(logcdf_upper, logcdf_lower)
+
+
+class MeasurableSwitchEncoding(MeasurableElemwise):
+    """A placeholder used to specify the log-likelihood for a encoded RV sub-graph."""
+
+    valid_scalar_types = (Switch,)
+
+
+measurable_switch_encoding = MeasurableSwitchEncoding(scalar_switch)
+
+
+@node_rewriter(tracks=[switch])
+def find_measurable_switch_encoding(
+    fgraph: FunctionGraph, node: Node
+) -> Optional[List[MeasurableSwitchEncoding]]:
+    rv_map_feature: Optional[PreserveRVMappings] = getattr(fgraph, "preserve_rv_mappings", None)
+
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    valued_rvs = rv_map_feature.rv_values.keys()
+
+    switch_condn, *components = node.inputs
+
+    # broadcasting of switch condition is not supported
+    if switch_condn.ndim != 0:
+        if any(switch_condn.type.broadcastable):
+            return None
+
+    if rv_map_feature.request_measurable([switch_condn]) != [switch_condn]:
+        return None
+        # this automatically checks the measurability of the switch condition and converts switch to MeasurableSwitch
+
+    measurable_comp_idx = next(
+        (
+            idx
+            for idx, component in enumerate(components)
+            if check_potential_measurability([component], valued_rvs)
+        ),
+        -1,
+    )
+
+    # If at least one of the branches is measurable
+    if measurable_comp_idx != -1:
+        measurable_component = components[measurable_comp_idx]
+
+        # broadcasting of the measurable component is not supported
+        if measurable_component.ndim != 0 and any(measurable_component.type.broadcastable):
+            return None
+
+        if not compare_measurability_source([switch_condn, measurable_component], valued_rvs):
+            return None
+
+        measurable_inputs = rv_map_feature.request_measurable(components)
+        # Maximum one branch allowed to be measurable
+        if len(measurable_inputs) > 1:
+            return None
```

Comment:
This has to be checked some other way than through `request_measurable`, as one input could become measurable before the other, and the check `>1` would not be triggered accidentally